dbt/models/default/default.vw_pin_sale.sql

```diff
@@ -249,93 +244,176 @@ sales_val AS (
         sf.run_id AS sv_run_id,
         sf.version AS sv_version
     FROM
-        {{ source('sale', 'flag') }}
-            AS sf
+        {{ source('sale', 'flag') }} AS sf
     INNER JOIN max_version_flag AS mv
         ON sf.meta_sale_document_num = mv.meta_sale_document_num
         AND sf.version = mv.max_version
+),
+
+combined_sales AS (
+    SELECT
+        COALESCE(uq_sales.pin, md_sales.pin) AS pin,
+        COALESCE(uq_sales.year, md_sales.year) AS year,
+        COALESCE(uq_sales.township_code, tc.township_code) AS township_code,
+        COALESCE(uq_sales.nbhd, tc.nbhd) AS nbhd,
+        COALESCE(uq_sales.class, tc.class) AS class,
+        COALESCE(
+            CASE
+                WHEN md_sales.sale_date < '2021-01-01' THEN md_sales.sale_date
+                WHEN uq_sales.sale_date >= '2021-01-01' THEN uq_sales.sale_date
+            END,
+            md_sales.sale_date,
+            uq_sales.sale_date
+        ) AS sale_date,
+        CASE
+            WHEN md_sales.sale_date < '2021-01-01' THEN TRUE
+            WHEN uq_sales.sale_date >= '2021-01-01' THEN FALSE
+            WHEN md_sales.sale_date IS NOT NULL THEN TRUE
+            WHEN uq_sales.sale_date IS NOT NULL THEN FALSE
+        END AS is_mydec_sale,
```

Comment:
Is the `is_mydec_date` logic sufficient here? Or should we be doing another branch of conditional logic for both
- no ias sale date after 2021
- no mydec sale date before 2020

Reply:
`CASE
    WHEN (uq_sales.year < '2021' OR uq_sales.sale_date IS NULL) AND md_sales.sale_date IS NOT NULL THEN TRUE
    WHEN (uq_sales.year >= '2021' OR md_sales.sale_date IS NULL) AND uq_sales.sale_date IS NOT NULL THEN FALSE
    END AS is_mydec_date`

If your logic aligns with this, and assuming I'm thinking correctly, it should be good.

Reply:
Yours is much more readable, I'll add it